lib/puppet/provider/package/pkgng.rb

```diff
@@ -0,0 +1,145 @@
+require 'puppet/provider/package'
+
+Puppet::Type.type(:package).provide :pkgng, :parent => Puppet::Provider::Package do
+  desc "A PkgNG provider for FreeBSD and DragonFly."
+
+  commands :pkg => "/usr/local/sbin/pkg"
+
+  confine :operatingsystem => [:freebsd, :dragonfly]
+  confine :pkgng_enabled => :true
+
+  defaultfor :operatingsystem => :freebsd
+  defaultfor :pkgng_enabled => :true
```

Comment:
If this is going into Puppet for 4.0, it should be in Facter 2.4 so it needs to target puppetlabs/facter, with a CFACT ticket added for tracking Facter 3. This seems reasonable as a fact.

Can this be enabled by configuration via Puppet? If so, how do we handle configuration changes that change the facts reported by the system?